src/shared/lib/helpers/validation/validation.ts

```diff
@@ -1,24 +1,38 @@
-export const emailChecking = (email: string, setter: (message: string) => void) => {
-    if (!email.trim()) {
-        setter('Field cannot be empty');
-        return false;
+export class FormValidator {
+    static isNotEmpty(value: string, setter: (message: string) => void): boolean {
+        if (!value.trim()) {
+            setter('Field cannot be empty');
+            return false;
+        }
+        return true;
     }
-    const reg = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
-    if (!reg.test(email)) {
-        setter('Must be an email');
-        return false;
+
+    static isEmail(value: string, setter: (message: string) => void): boolean {
+        const reg = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+        if (!reg.test(value)) {
+            setter('Must be an email');
+            return false;
+        }
+        return true;
+    }
+
+    static isPassLengthCorrect(value: string, setter: (message: string) => void): boolean {
+        if (value.trim().length < 6) {
+            setter('Password must be at least 6 characters');
+            return false;
+        }
+        return true;
     }
-    return true;
-};
 
-export const passChecking = (pass: string, setter: (message: string) => void) => {
-    if (!pass.trim()) {
-        setter('Field cannot be empty');
-        return false;
+    static passChecking(pass: string, setter: (message: string) => void): boolean {
+        return this.isNotEmpty(pass, setter) && this.isPassLengthCorrect(pass, setter);
     }
-    if (pass.length < 6) {
-        setter('Password must be at least 6 characters');
-        return false;
+
+    static emailChecking(email: string, setter: (message: string) => void): boolean {
+        return this.isNotEmpty(email, setter) && this.isEmail(email, setter);
+    }
+
+    static performerChecking(performer: string, setter: (message: string) => void): boolean {
+        return !performer.trim() || this.isEmail(performer, setter);
     }
```

Comment:
seems, it can be one function `emailChecking` with an additional parameter which indicate whether an empty value allowed or not

Reply:
Fixed